src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -630,6 +770,7 @@ void Terminal::ClearSelection()
 {
     _selection = std::nullopt;
     _selectionMode = SelectionInteractionMode::None;
+    _isTargetingUrl = false;

```

Comment:
who resets `_isTargetingUrl` when somebody (1) makes a mouse selection? or (2) _extends an automatic URL search selection by shift-clicking?_

Reply:
It's only set in `SelectHyperlink()` (you tabbed to the hyperlink).

It's then reset when you...
- clear the selection
- enter/exit mark mode (I guess this one is unnecessary)
- use keyboard selection to modify the selection

So if you make a mouse selection (or extend it via shift+click), we're no longer in mark mode so ctrl+enter doesn't work.

_That said_, if we choose a different path [here](https://github.com/microsoft/terminal/pull/13405#discussion_r919514369), `_isTargetingUrl`'s value does matter and should be reset there. Heck, in general, it's probably just good practice to update `_isTargetingUrl` then so I might as well do it.